File: little-bee/controller/router/device_type.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_DEVICE_TYPE = "/config/device-type"
const URL_ROUTER_DEVICE_TYPE_BY_ID = "/config/device-type/:device-type-id"
const URL_ROUTER_DEVICE_TYPE_QUERY = "/config/device-type/query"
const URL_ROUTER_DEVICE_TYPE_ALL = "/config/device-type/all"

func DeviceTypeRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_DEVICE_TYPE, config.CreateDeviceType)
	r.GET(URL_ROUTER_DEVICE_TYPE_BY_ID, config.RetrieveDeviceType)
	r.PUT(URL_ROUTER_DEVICE_TYPE, config.UpdateDeviceType)
	r.DELETE(URL_ROUTER_DEVICE_TYPE_BY_ID, config.DeleteDeviceType)
	r.PUT(URL_ROUTER_DEVICE_TYPE_BY_ID, config.DeleteDeviceTypeWithFlag)
	r.POST(URL_ROUTER_DEVICE_TYPE_QUERY, config.QueryDeviceType)
	r.GET(URL_ROUTER_DEVICE_TYPE_ALL, config.GetAllDeviceTypes)
	return r
}

File: frontend/console/src/api/device-type.js
import Axios from "@/http";

export function createDeviceType(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/device-type', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveDeviceType(deviceTypeId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/device-type/${ deviceTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateDeviceType(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/device-type', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteDeviceType(deviceTypeId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/device-type/${ deviceTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteDeviceTypeWithFlag(deviceTypeId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/device-type/${ deviceTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllDeviceTypes() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/device-type/all').then(res => {
            resolve(res)
        })
    })
}

export function queryDeviceType (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/device-type/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}